### PR TITLE
[docs] add intro content from training module

### PR DESCRIPTION
**Pull request by angelinalg**
## Why are these changes needed?
The docs can use some high level introductory material. The training modules have very nice content to leverage! I included some features of Ray and a description of the libraries.

This is a Work In Progress [WIP] to get some early feedback before fleshing out more.
cc: @emmyscode

### CODE REVIEW

**Comment by pcmoritz** [comment]
The content looks great, have you considered putting it into https://docs.ray.io/en/latest/index.html which is the canonical landing page for the Ray documentation? And then link it from "Getting started" with a line like "If you want to learn more about what Ray is and what you can do with it, see [here]" at the top? That would make sure that content is not duplicated and the "Getting started" guide keeps getting to the point quickly, yet giving people that don't know about Ray the opportunity to learn :)

**Comment by angelinalg** [comment]
You are so right, @pcmoritz. I broke the cardinal rule of mixing content styles. I will rework!

**Comment by angelinalg** [comment]
Summary of changes:
1. Moved overview content from training to the home page.
2. Moved the "Why Ray?" to the intro paragraph, and "What is Ray? before "What can you do" to improve the flow of content. 
3. Removed the "What documentation resource is right for you?" because with the recent IA changes, the navigation is intuitive.
4. Moved the logo further down so that user doesn't need to scroll as much to get to content.
5. Tightened up the text for the cards in "What can you do".
Before:
Screenshot 2023-02-07 at 11 39 24 AM
After:
Screenshot 2023-02-07 at 11 39 10 AM

**Comment by richardliaw** [comment]
this is... beautiful....

File: doc/source/ray-overview/what-is-ray.md
Comment (maxpumperla) [inline doc/source/ray-overview/what-is-ray.md:3]: @angelinalg I think you left this document in by accident, correct? All changes should be in  `index.md`.

File: doc/source/index.md
Comment (maxpumperla) [inline doc/source/index.md:205]: you moved the header and all logos to the bottom (see rendered page). I don't think that's what we want. The badges we can discuss, but a prominent logo should still be featured up top.

File: doc/source/index.md
Comment (maxpumperla) [inline doc/source/index.md:164]: I think the image for this one is really hard to see. Maybe we can just have parts of it, or mention that you can "click to enlarge" or so.

File: doc/source/index.md
Comment (maxpumperla) [inline doc/source/index.md:166]: Probably not for this PR, but I think we're already using this plugin https://github.com/sphinx-contrib/youtube to embed yt videos directly. Having more multi-modal content is important, and I think going through our yt channel and take inventory in terms of what we can cross-link on our docs could be extremely valuable.

**Review by maxpumperla — DISMISSED** [review]
@angelinalg really like this! just left a couple of comments to address.

File: doc/source/ray-overview/what-is-ray.md
Comment (maxpumperla) [reply doc/source/ray-overview/what-is-ray.md:3]: once you remove this doc, the CI build should also pass.

File: doc/source/index.md
Comment (emmyscode) [reply doc/source/index.md:164]: This diagram has a lot going on. A new version is in the works for upcoming serving tutorials, but in the meantime, I can forward along some alternatives.

Comment (angelinalg) [reply doc/source/index.md:164]: I'll add a note to "Click image to enlarge." in the caption. @emmyscode, I'd love to update it when the new version is available. Thanks!

File: doc/source/index.md
Comment (angelinalg) [reply doc/source/index.md:166]: Good to know. I agree that we could do more cross-linking in general. Thanks!

File: doc/source/ray-overview/what-is-ray.md
Comment (angelinalg) [reply doc/source/ray-overview/what-is-ray.md:3]: Thanks for catching that, @maxpumperla! I'll take care of it.

File: doc/source/index.md
Comment (angelinalg) [reply doc/source/index.md:205]: Let me share why I removed it from the top. I thought that it was generally a bad user experience to have to scroll through graphics to get to content. The logo is also in the upper left hand corner, so the larger logo at the top is redundant.

I referenced other docs home pages to check my bias:

- https://docs.databricks.com/
- https://modal.com/docs/guide (their docs have been coming up in conversations recently)
- https://stripe.com/docs
- https://www.twilio.com/docs
- https://platform.openai.com/docs/introduction
- https://docs.grid.ai/

Those examples all have the logo in the upper left corner. WDYT, @maxpumperla? Let me know if you think my examples are too skewed.

Comment (simran-2797) [reply doc/source/index.md:206]: Hey! I like it! I do agree that we do not need the logo on top of the page since we already have the Ray logo on the horizontal navigation bar. We also have the text "Ray 3.0.0" on the left vertical navigation, so putting the Ray logo again in the page might be redundant.

Comment (angelinalg) [reply doc/source/index.md:206]: Yes, I had the same feeling of splitting out the ML Workloads into a separate page. Thanks, @simran-2797! I'll file an issue to "freeze" the top nav.

**Comment by pcmoritz** [comment]
This looks great to me now, thanks a lot for doing this -- let me know when it is ready to give a little more detailed feedback. I'd consider putting the badges back on top so people can find our slack etc. quickly but up to you.

**Comment by angelinalg** [comment]
Addressed comments from @maxpumperla, @Wendi-anyscale, and @simran-2797, and input from @richardliaw.
1. Split home page into a landing page (index.md) and an overview page (ray-overview/index.md), and added "Overview" to the TOC under "Ray".
2. Moved "Common ML workloads" into a separate page and linked to it from the "Overview" page. (Having a problem making "Common ML workloads" a child page of "Overview". Any tips would be appreciated!)
3. Replaced intro paragraph of "Overview" page (which was the intro paragraph of the original home page), by an intro that ML practitioners can relate more to.

Trying to get this signed off today to be included in 2.3 so please let me know if you have any blocking comments! Thanks!

**Comment by angelinalg** [comment]
@maxpumperla, I had trouble making "Common ML workloads" a child page of "Overview". After running some experiments, it appears that the problem might be that "Overview" is an MD file. I didn't know if you have a preferred MD>RST convertor, so I left it on the top-level. If you can easily convert it, or somehow figure out how an MD file can take a child page, would you mind doing it for me? Thanks!

**Comment by angelinalg** [comment]
@cadedaniel, please include this in 2.3 after it gets approved. TY!

File: doc/source/index.md
Comment (maxpumperla) [reply doc/source/index.md]: @angelinalg ok, understood. removing the logo is fine. I just thought it looked silly to put it at the bottom - better to remove altogether.

**Comment by maxpumperla** [comment]
@angelinalg your problems, I believe come from the fact that you moved contents of a file from the base folder to `ray-overview`. That means all those relative references, like `data/datasets.rst` can't be found anymore (obv. just use `../data/datasets.rst` etc.).

Personally, I'm a proponent of myst markdown over rst, as the latter is just not a great format to read or write. If you have to know, rst to myst conversion is best done with [this python lib](https://pypi.org/project/rst-to-myst/), and the converse you could do with [pandoc](https://gist.github.com/zaiste/77a946bbba73f5c4d33f3106a494e6cd). Just note that that won't solve your problems, you have to apply the fixes as indicated above.

Makes sense? We can't merge this before all doc tests pass.

**Comment by angelinalg** [comment]
@pcmoritz, the badges are back up on top of the home page. Let me know if you have any more blocking issues! Thx!

File: doc/source/index.md
Comment (angelinalg) [reply doc/source/index.md]: I removed the logo. Thanks, @maxpumperla.

**Comment by Wendi-anyscale** [comment]
Love the changes! These definitely help fix the issues we talked about.
Would it be great to use the same format for these 2 paragraphs and call out the persona in the front?
e.g.

- For Data scientist
- For Distributed system engineers…
Screenshot 2023-02-14 at 12 00 23 PM
 and machine learning practitioners….

File: doc/source/_toc.yml
Comment (richardliaw) [inline doc/source/_toc.yml:10]: image
this ends up rendering a bit oddly (it wasn't clear that the header was clickable)

Comment (angelinalg) [reply doc/source/_toc.yml:10]: You mean the carrot didn't show up?

Comment (richardliaw) [reply doc/source/_toc.yml:10]: no, i mean that it wasnt clear to me that you could actually click the overview page

Comment (angelinalg) [reply doc/source/_toc.yml:10]: Ah. Yes, this is a source of debate, whether we should put the "Common ML workloads" page at the top level. Max was trying to keep the top level short because it was really bloated. Are you suggesting that I pull it up to the top level?

Comment (angelinalg) [reply doc/source/_toc.yml:10]: Maybe what's not clear is that the `Overview` page links to the `Common ML workloads` page: 
Screenshot 2023-02-14 at 5 36 31 PM

Comment (angelinalg) [reply doc/source/_toc.yml]: I pulled the Workloads page to the top level in the side nav.
Screenshot 2023-02-14 at 6 38 56 PM

**Comment by richardliaw** [comment]
Seems like there's some conflicts?

**Review by pcmoritz — APPROVED** [review]
This looks great, thanks for doing it!

**Pull request merged**
